test/Microsoft.ML.Core.Tests/UnitTests/TestCSharpApi.cs

```diff
@@ -528,6 +528,89 @@ public void TestCrossValidationMacroWithMultiClass()
                     }
                     Assert.Equal(0, rowCount);
                 }
+
+                var warnings = experiment.GetOutput(crossValidateOutput.Warnings);
+                using (var cursor = warnings.GetRowCursor(col => true))
+                    Assert.False(cursor.MoveNext());
+            }
+        }
+
+        [Fact]
+        public void TestCrossValidationMacroMultiClassWithWarnings()
+        {
+            var dataPath = GetDataPath(@"Train-Tiny-28x28.txt");
+            using (var env = new TlcEnvironment(42))
+            {
+                var subGraph = env.CreateExperiment();
+
+                var nop = new ML.Transforms.NoOperation();
```

Comment:
Using entrypoints graph is fine for now in C# but please consider not using it and instead call the low level C# APIs for component initialization from now on. We will be getting rid of CSharpCodeGenerator as there is no need to use C# code generated APIs when low level C# APIs already exist. Refer to issue #371